nemo/collections/common/parts/preprocessing/manifest.py

```diff
@@ -75,9 +84,44 @@ def item_iter(
         logging.debug('Using manifest file: %s', str(manifest_file))
         cached_manifest_file = DataStoreObject(manifest_file).get()
         logging.debug('Cached at: %s', str(cached_manifest_file))
+
+        if shard_strategy is not None and world_size > 1:
+            if shard_strategy == 'scatter':
+                # Read only necessary manifest lines
+                logging.debug('Partitioning manifest: global rank = %d, world size = %d', global_rank, world_size)
+
+                # Get number of lines in the current manifest
+                num_lines = get_num_lines(cached_manifest_file)
+                logging.debug('Found %d lines in manifest %s', num_lines, cached_manifest_file)
+
+                # Determine start and end lines in the manifest.
+                # This corresponds to splitting the manifest in audio_to_text.py::expand_audio_filepaths
+                begin_idx = (num_lines // world_size) * global_rank
```

Comment:
I don't fully understand how this works. Are we trying to select all the lines that have shards selected for the current rank? If so, is the assumption here that they always go in sequence and each shard has the same number of lines? I'm just not sure if this is always the case (or if we should assume that it's going to always be the case in the future).

Reply:
And in any case, I think it's very important to have some tests for this functionality, as it might be very easy to add hard-to-notice bugs here by ignoring part of the data